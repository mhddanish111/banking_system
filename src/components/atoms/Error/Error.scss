@import "../../../styles/colors.scss";
@import "../../../styles/boarder.scss";

.error-container {
  display: flex;
  flex-direction: column;
  color: $color-red-50;
  width: 100%;
  text-align: center;
  background-color: $color-red-10;
  border: $border-dark;
  border-radius: 10px;
  box-shadow: 1px 1px 10px rgba(161, 182, 253, 0.825);
  padding: 10px;
}
