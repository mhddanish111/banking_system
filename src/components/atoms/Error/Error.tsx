import React from 'react';
import './Error.scss';

type ErrorProps = {
  message: String;
};
const Error: React.FC<ErrorProps> = ({ message }) => {
  return (
    <div className="error-container">
      <span>Something went wrong. Please try again</span>
      {message && <span>{message}</span>}
    </div>
  );
};

export default Error;
