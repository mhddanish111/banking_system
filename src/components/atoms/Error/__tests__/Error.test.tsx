import React from 'react';
import { shallow } from 'enzyme';
import Error from '../Error';

describe('<Error />', () => {
  const props = {
    message: 'Something went wrong',
  };
  const wrapped = shallow(<Error {...props} />);
  it('should render the Error Component correctly', () => {
    expect(wrapped).toMatchSnapshot();
  });
});
