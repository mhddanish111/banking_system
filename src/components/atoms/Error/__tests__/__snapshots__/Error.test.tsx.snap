// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> should render the Error Component correctly 1`] = `
<div
  className="error-container"
>
  <span>
    Something went wrong. Please try again
  </span>
  <span>
    Something went wrong
  </span>
</div>
`;
